service/cvpartner.py

```diff
@@ -214,11 +200,39 @@ def __put_custom_tag(self, url, entity):
         logger.debug(entity["payload"])
         req = requests.put(url, headers=headers, json=entity["payload"])
         if req.status_code != 200:
-            logger.error(
-                "Unexpected response status code: %d with response text %s" % (req.status_code, req.text))
-            abort(req.status_code, req.json())
+            req = self.check_error(req, cv_url, headers, 'put', 'json', entity["payload"])
         return str(req.status_code)
 
+    def check_error(self, req, url, headers, method, json = None, data = None):
+        if req.status_code == 429:
+            return self.recursive_request(url, headers, req.headers.get('Retry-After'), method, json, data)                        
+        else:
+            logger.error("Unexpected response status code: %d with response text %s" % (req.status_code, req.text))
+            raise AssertionError("Unexpected response status code: %d with response text %s" % (req.status_code, req.text))        
+
+    def recursive_request(self, url, headers, retry_after, method, json = None, data = None):
+        logger.info("Sleeping for %.2f seconds" % float(retry_after))
+        sleep(float(retry_after))
+        if method == 'get':
```

Comment:
suggestion for thie whole if-else block
```suggestion
        if method in('get', 'post', 'put'):
            req = requests.request(method=method, url=url, headers=headers, json=json, data=data)
```

Reply:
nice, updated